test/k4FWCoreTest/options/CheckOutputFiles.py

```diff
@@ -118,3 +119,13 @@ def check_collections(filename, names):
     "output_k4test_exampledata_functional_mix_iosvc.root",
     mix_collections,
 )
+
+f = ROOT.TFile.Open("functional_producer_hist.root")
+for i in range(2):
+    if (
+        str(f.GetListOfKeys()[i])
```

Comment:
Is this guaranteed to be always ordered?

Reply:
In the current test I left multithreading disabled not to have too many multithreaded jobs running at same time since `ctest` doesn't know about this but after running it several times I never got the other ordering. I think the histograms may be registered when the class is created and in that case it would always be ordered but I'm not sure.

Reply:
The alternative would be to simply make a set of the names in the files and the expected ones and do a set comparison to make sure everything is there to avoid any ordering issues.